Admin/PageAdmin.php

```diff
@@ -449,8 +449,15 @@ protected function configureSideMenu(MenuItemInterface $menu, $action, AdminInte
 
         if (!$this->getSubject()->isHybrid() && !$this->getSubject()->isInternal()) {
             try {
-                $menu->addChild('view_page',
-                    array('uri' => $this->getRouteGenerator()->generate('page_slug', array('path' => $this->getSubject()->getUrl())))
+                $url = $this->getSubject()->getUrl();
+                $relativePath = $page->getSite()->getRelativePath();
+                if (!empty($relativePath)) {
+                    $url = $relativePath.$url;
```

Comment:
That won't work if `getUrl()` returns an absolute url. If it can't, `getUrl()` is badly named, but you can still name your variable correctly (using `$path` instead of `$url`). Likewise `$relativePath` should probably be named `$siteRelativePath`, that way will end up here with `$path = $siteRelativePath . $path`.